client/manager/manager.go

```diff
@@ -192,3 +209,36 @@ func (m *ClientManager) addClientCredentials(cli *client.Client) error {
 	}
 	return nil
 }
+
+func validateClient(cli client.Client) error {
+	if cli.Public {
+		if len(cli.Metadata.RedirectURIs) > 0 {
+			return client.ErrorPublicClientRedirectURIs
+		}
+		if cli.Metadata.ClientName == "" {
+			return client.ErrorPublicClientMissingName
+		}
+		cli.Metadata.RedirectURIs = []url.URL{
```

Comment:
well, metadata.Valid() does a whole bunch of other checks that we'd like validated right?